client/my-sites/domains/domain-management/edit-contact-info/form-card.jsx

```diff
@@ -424,6 +391,7 @@ class EditContactInfoFormCard extends Component {
 				this.state.transferLock,
 				this.state.updateWpcomEmail
 			);
+			this.updateWpcomEmail( newContactDetails, this.state.updateWpcomEmail );
```

Comment:
I think it's a little confusing that even though the `onSubmitButtonClick` prop has returned `'cancel'` the component isn't immediately canceling it's own processing.

The doc comment for the `onSubmitButtonClick` prop says returning `'cancel'` will cancel the default handling.

Given that `onSubmitButtonClick` is passed the value of `updateWpcomEmail` I think the bulk update page's submit button handler should be able to decide whether or not to update the wpcom email and dispatch the `createUserEmailPendingUpdate` action itself. By returning `'cancel'` it's saying we've handled everything that needs to be handled.

It doesn't change anything about how this PR runs, but I think it'll be clearer when someone revisits this code down the track.

Reply:
Do you mean you'd like to lift this decision to the component that is using `EditContactInfoFormCard`? The problem is there are two instances, so we'd need to call this in two different places. Plus, there's some logic that prevents or allows the calling of the update settings endpoint too.

I think one thing that can be done is to move the `this.updateWpcomEmail();` call outside the `onSubmitButtonClick` conditional:

```diff
diff --git a/client/my-sites/domains/domain-management/edit-contact-info/form-card.jsx b/client/my-sites/domains/domain-management/edit-contact-info/form-card.jsx
index 540dbd73ad..5dbe0d2384 100644
--- a/client/my-sites/domains/domain-management/edit-contact-info/form-card.jsx
+++ b/client/my-sites/domains/domain-management/edit-contact-info/form-card.jsx
@@ -385,13 +385,15 @@ class EditContactInfoFormCard extends Component {
 	};
 
 	handleSubmitButtonClick = ( newContactDetails ) => {
+		this.updateWpcomEmail( newContactDetails, this.state.updateWpcomEmail );
+
 		if ( this.props.onSubmitButtonClick ) {
 			const result = this.props.onSubmitButtonClick(
 				newContactDetails,
 				this.state.transferLock,
 				this.state.updateWpcomEmail
 			);
-			this.updateWpcomEmail( newContactDetails, this.state.updateWpcomEmail );
+
 			if ( result === 'cancel' ) {
 				return;
 			}

```

Like this, it will be clearer that the email update is an independent action to the submit button click handler. And well, it is.

Reply:
I see. Maybe my original idea of allowing components who use `EditContactInfoFormCard` to cancel processing just isn't a good fit anymore.

My intention behind doing it this way was that I didn't want to make a lot of changes to `EditContactInfoFormCard`. I thought I could mostly leave it as it was and then do any of the things needed for bulk updates in the parent component. So "we'd need to call this in two different places" as you say was an intentional trade off.

But as we've had to add more customisations to `EditContactInfoFormCard` (it even has a `bulkEdit` prop now) that intention doesn't make much sense any more. We could have all the logic in `handleSubmitButtonClick` now, calling the bulk update endpoints. And then it'd obviously just call `updateWpcomEmail` too.

Disabling the submit button while bulk updating also doesn't work correct at the moment. It'd fix that problem too.

So I take it back, I don't think we need to move the `updateWpcomEmail` out of this component any more.
We also don't have to make a refactor in this PR either. But it's good to figure it out in my head :)